client/lib/domains/dns/index.js

```diff
@@ -45,8 +46,8 @@ function isValidDomain( name ) {
 	return /^([a-z0-9-_]{1,63}\.)*[a-z0-9-]{1,63}\.[a-z]{2,63}$/i.test( name );
 }
 
-function isValidName( name, type ) {
-	if ( isNameEmptyAndItsAllowed( name, type ) ) {
+function isValidName( name, type, domainName ) {
+	if ( isRootDomainAndItsAllowed( name, type, domainName ) ) {
```

Comment:
I wanted to stress the "allowed" part - the root domain might be in a valid format, but it might not be _allowed_ for the given record type (for example, we allow root domains for TXT but not for CNAME). I wanted the name of the function to hint at that difference: `isRootDomain` means we are dealing with a root domain and `AndItsAllowed` means that we are checking if it's allowed for the given record type.